### PR TITLE
more test coverage of ymax-planner

### PR DESCRIPTION
**Pull request by turadg**
closes: https://github.com/Agoric/agoric-private/issues/563

### CODE REVIEW

**Comment by turadg** [comment]
@gibson042 this is a bit of AI slop test coverage. Which parts do you think would be useful?

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+import { createMockPendingTxOpts } from './mocks.ts';
+
+const marshaller = boardSlottingMarshaller();
+const depositBrand = Far('mock brand') as Brand<'nat'>;
+
+// ============================================================================
+// 1. Vstorage Data Parsing Edge Cases
+// ============================================================================
+
+test('parseStreamCell handles various blockHeight formats', t => {
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:28]: These `parseStreamCell` and `parseStreamCellValue` tests are reasonable, but I'd like to see them in a file more obviously associated with vstorage-utils (which I expect to move elsewhere).

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+
+  const parsed = parseStreamCellValue(cell, 0, 'path');
+  t.deepEqual(JSON.parse(parsed), complexValue);
+});
+
+// ============================================================================
+// 2. Pending Transaction Event Processing
+// ============================================================================
+
+test('processPendingTxEvents skips non-pending transactions', async t => {
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:205]: These tests should be reviewed by @frazarshad and/or @rabi-siddique, and likewise probably merit their own file.

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+  await processPendingTxEvents(events, mockHandler, createMockPendingTxOpts());
+
+  // Should only process the last value (index -1)
+  t.is(handledTxs.length, 1);
+  t.is(handledTxs[0]?.type, TxType.GMP);
+  t.is(handledTxs[0]?.amount, 2000000n);
+});
+
+// ============================================================================
+// 3. makeVstorageEvent Function Tests
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:386]: These don't seem useful, we're already covered by a combination of TypeScript and higher-level testing.

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+
+  const streamCell = JSON.parse(streamCellJson);
+  const capData = JSON.parse(streamCell.values[0]);
+  const unmarshalled = marshaller.fromCapData(capData);
+
+  t.deepEqual(unmarshalled, complexValue);
+});
+
+// ============================================================================
+// 4. EVM Log Parsing Edge Cases
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:476]: I don't think these are useful, but that's more a question for @frazarshad and @rabi-siddique. But if they do remain, I think they belong in their own file.

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+
+  const multicallStatusSig = id('MulticallStatus(string,bool,uint256)');
+  t.is(
+    multicallStatusSig,
+    '0x0c32fa99b6e87f667e61e6a78d88a7906d8ec6dc6e6c8fb93b2d0e99d18de504',
+  );
+});
+
+// ============================================================================
+// 5. Path Encoding Edge Cases
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:545]: These tests are not valuable.

File: services/ymax-planner/test/blockchain-parsing.test.ts [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+  const parts = encodedKey.split('\x00');
+  const count = parseInt(parts[0], 10);
+  t.is(count, 20);
+
+  const decodedPath = parts.slice(1, count + 1).join('.');
+  t.is(decodedPath, path);
+});
+
+// ============================================================================
+// 6. Number Encoding Edge Cases
```
Comment (gibson042) [inline services/ymax-planner/test/blockchain-parsing.test.ts:601]: These tests are not valuable.

File: services/ymax-planner/test/plan-withdraw.property.test.ts [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+  .chain(record => {
+    const total = Object.values(record).reduce((sum, n) => sum + n, 0);
+    return fc.record({
+      balances: fc.constant(record),
+      withdrawal: fc.integer({ min: 0, max: total }),
+    });
+  });
+
+testProp(
+  'computeWeightedTargets distributes withdraw proportionally and keeps the remainder on the heaviest weight',
```
Comment (gibson042) [inline services/ymax-planner/test/plan-withdraw.property.test.ts:33]: This is a good concept, but is not being tested correctly. I think the appropriate input would be balances of dynamic cardinality and minimum amount of e.g. 1000 uusdc and appropriate assertions would be that the relative share after withdrawal shifts by no more than e.g. 1%.

File: services/ymax-planner/test/plan-withdraw.property.test.ts [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+      const expectedValue = key === maxKey ? share + remainder : share;
+      t.is(targetValue, expectedValue);
+    }
+  },
+);
+
+const allKeys = ['Aave_Avalanche', 'Compound_Arbitrum', 'USDN'] as string[];
+
+testProp(
+  'computeWeightedTargets zeros current balances that lost their target allocation',
```
Comment (gibson042) [inline services/ymax-planner/test/plan-withdraw.property.test.ts:87]: I can't determine what this is intended to test.

File: services/ymax-planner/test/plan-withdraw.property.test.ts [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    );
+
+    for (const key of missingWithValue) {
+      t.true(key in draft);
+      t.is(draft[key]!.value, 0n);
+    }
+  },
+);
+
+test('planWithdrawFromAllocations zeros hubs and tops up cash before planning', async t => {
```
Comment (gibson042) [inline services/ymax-planner/test/plan-withdraw.property.test.ts:138]: "planWithdrawFromAllocations zeros hubs" is another good concept with a poor implementation.

File: services/ymax-planner/test/plan-withdraw.property.test.ts [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+  );
+
+  t.deepEqual(steps, ['stub-step']);
+  const [{ target, current }] = planned;
+  t.is(target['@Arbitrum']?.value, 0n);
+  t.is(target['<Cash>']?.value, 60n);
+  t.is(current['<Cash>']?.value, 10n);
+});
+
+test('planWithdrawFromAllocations rejects negative allocation weights', async t => {
```
Comment (gibson042) [inline services/ymax-planner/test/plan-withdraw.property.test.ts:171]: Such input validation is good; I'd like to see it generalized and simplified.
```ts
test(
  'computeWeightedTargets rejects negative target allocation weights',
  testComputeWeightedTargetsRejection,
  {
    targetAllocation: { USDN: -1n, Aave_Arbitrum: 10n },
  },
  { message: /allocation weight.*must be a Nat/ },
);

test(
  'computeWeightedTargets requires positive total target allocation weight',
  testComputeWeightedTargetsRejection,
  {
    targetAllocation: { USDN: 0n, Aave_Arbitrum: 0n },
  },
  { message: /allocation weight.*must be a Nat/ },
);
```

File: services/ymax-planner/test/timeout-edge-cases.test.ts
Comment (gibson042) [inline services/ymax-planner/test/timeout-edge-cases.test.ts:1]: Best reviewed by @frazarshad and/or @rabi-siddique.

File: services/ymax-planner/test/input-validation.test.ts [hunk 11]
```diff
@@ -0,0 +1,745 @@
...
+const depositBrand = Far('mock brand') as Brand<'nat'>;
+const feeBrand = Far('fee brand (BLD)') as Brand<'nat'>;
+const makeDeposit = (value: bigint) => AmountMath.make(depositBrand, value);
+
+// ============================================================================
+// 1. Invalid Input Validation
+// ============================================================================
+
+test('pickBalance returns undefined for empty balances', t => {
+  const usdc = {
```
Comment (gibson042) [inline services/ymax-planner/test/input-validation.test.ts:40]: This should be defined at top-level scope along with `depositBrand` (ideally with a shared `mockAsset`/`fakeAssetInfo`/`makeAssetInfo` helper), allowing for simplification:
```ts
test('pickBalance returns undefined for empty balances', t => {
  t.is(pickBalance(undefined, usdcAsset), undefined);
  t.is(pickBalance([], usdcAsset), undefined);
});

test('pickBalance returns undefined for missing denom', t => {
  const balances = [
    { amount: '50', denom: 'ibc/5678' },
    { amount: '123', denom: 'ubld' },
  ];
  t.is(pickBalance(balances, usdcAsset), undefined);
});
```

File: services/ymax-planner/test/input-validation.test.ts [hunk 12]
```diff
@@ -0,0 +1,745 @@
...
+  };
+
+  const balances = [{ amount: 'not-a-number', denom: 'ibc/1234' }];
+
+  t.throws(() => pickBalance(balances, usdc), {
+    message: /Cannot convert/,
+  });
+});
+
+test('parseStreamCell throws on malformed JSON', t => {
```
Comment (gibson042) [inline services/ymax-planner/test/input-validation.test.ts:88]: As with blockchain-parsing.test.ts, `parseStreamCell` and `parseStreamCellValue` tests should move to a file more obviously associated with vstorage-utils.

File: services/ymax-planner/test/input-validation.test.ts [hunk 13]
```diff
@@ -0,0 +1,745 @@
...
+
+  // Check that one of the errors mentions invalid address
+  t.true(
+    (error as AggregateError).errors.some(e =>
+      e.message.includes('Invalid CAIP-10'),
+    ),
+  );
+});
+
+test('planDepositToAllocations returns empty on missing targetAllocation', async t => {
```
Comment (gibson042) [inline services/ymax-planner/test/input-validation.test.ts:173]: These don't seem like input validation, but they are useful.

File: services/ymax-planner/test/input-validation.test.ts
Comment (gibson042) [inline services/ymax-planner/test/input-validation.test.ts:1]: There's lots of redundancy in this file; "input validation" cuts across too much.

**Comment by turadg** [comment]
No approval. I'll leave to mentioned devs for whether to pick commits to land.
